ts/features/itwallet/machine/eid/guards.ts

```diff
@@ -58,6 +59,7 @@ export const createEidIssuanceGuardsImplementation = (
       O.chain(x => O.fromNullable(x[EID_FISCAL_CODE_KEY]?.value)),
       O.map(t.string.decode),
       O.chain(O.fromEither),
+      O.map(x => x.replace(EID_FISCAL_CODE_PREFIX, "")),
```

Comment:
What about a regex to extract the fiscal code instead?

Reply:
How strict do we want it to be? It can get pretty ugly...

I came out with this: `([A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z]{1}[0-9LMNPQRSTUV]{3}[A-Z])`.

It simply matches patterns of `n` letters or numbers, considering numbers substitutions to handle "omocodia".